### PR TITLE
SASS-9021: Create Backend mongo- Stock dividends

### PR DESCRIPTION
**Pull request by AishaBi**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

[SASS-9021]

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -106,6 +106,7 @@ class FrontendAppConfig @Inject()(servicesConfig: ServicesConfig) extends AppCon
     servicesConfig.getString("microservice.services.income-tax-submission-frontend.exclude")
 
   lazy val tailoringEnabled: Boolean = servicesConfig.getBoolean("feature-switch.tailoringEnabled")
+  lazy val newStockDividendsServiceEnabled: Boolean = servicesConfig.getBoolean("feature-switch.newStockDividendsServiceEnabled")
```
Comment (ansahdavid) [inline app/config/AppConfig.scala:109]: Could this be renamed to something like backendSessionEnabled which describes what the feature switch is doing

File: conf/application.conf [hunk 2]
```diff
@@ -158,6 +158,7 @@ feature-switch {
 
    welshToggleEnabled = true
    tailoringEnabled = true
+   newStockDividendsServiceEnabled = false
```
Comment (ansahdavid) [inline conf/application.conf:161]: Could this be renamed to something like backendSessionEnabled which describes what the feature switch is doing

File: it/test/controllers/dividends/RedeemableSharesAmountControllerISpec.scala [hunk 3]
```diff
@@ -202,75 +227,130 @@ class RedeemableSharesAmountControllerISpec extends IntegrationTest with ViewHel
...
-          implicit val document: () => Document = () => Jsoup.parse(result.body)
+          titleCheck(errorPrefix(scenario.isWelsh) + expectedTitle, scenario.isWelsh)
+          errorAboveElementCheck(expectedErrorEmpty)
+          errorSummaryCheck(expectedErrorEmpty, Selectors.errorSummaryHref, scenario.isWelsh)
+        }
+
+        "the form is invalid with appWithStockDividends" which {
+          implicit lazy val app: Application = appWithStockDividends
+
+          lazy val result = postRedeemableSharesAmount(Seq("amount" -> "$$$"), app)
```
Comment (ansahdavid) [inline it/test/controllers/dividends/RedeemableSharesAmountControllerISpec.scala:317]: For consistency could this be renamed to `resultBackEnd`

File: it/test/controllers/dividends/RedeemableSharesStatusControllerISpec.scala [hunk 4]
```diff
@@ -204,56 +207,132 @@ class RedeemableSharesStatusControllerISpec extends IntegrationTest with ViewHel
...
         lazy val result = {
           dropStockDividendsDB()
-          authoriseAgentOrIndividual(scenario.isAgent)
-          urlPost(postURL, follow = false, headers = playSessionCookie(scenario.isAgent), body = Map("value" -> Seq("true")))
+          postRedeemableSharesStatus(Seq("value" -> "true"), app)
         }
-        result.status shouldBe SEE_OTHER
-        result.headers(HeaderNames.LOCATION).head shouldBe redeemableSharesAmountUrl
+        status(result) shouldBe SEE_OTHER
+        redirectLocation(result).value shouldBe redeemableSharesAmountUrl
```
Comment (ansahdavid) [inline it/test/controllers/dividends/RedeemableSharesStatusControllerISpec.scala:238]: Same here, and everywhere else if possible

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Review by Shihabur-Atos — APPROVED** [review]
Approved

**Pull request closed without merging**